src/common/Thread.h

```diff
@@ -34,6 +35,8 @@ class Thread {
   pid_t pid;
   int cpuid;
   std::string thread_name;
+  static std::mutex threadid_name_mutex;
+  static std::unordered_map<pthread_t, std::string> threadid_name;
```

Comment:
Doesn't this look like reinventing the _thread local storage_?

Reply:
It does look like it ... but I'm trying to help `Log::dump_recent()` find the thread name after a thread ceases to exist.
I'm also worried that a `pthread_t` may be recycled by the time `Log::dump_recent()` eventually calls `pthread_getname_np()`.

Reply:
I think I understand what's your intention. My point is recording in TLS would happen when the thread is still alive, not at the `dump_recent()` stage.